designer/server/src/main/scala/pl/touk/nussknacker/ui/process/migrate/RemoteEnvironment.scala

```diff
@@ -205,15 +222,48 @@ trait StandardRemoteEnvironment extends FailFastCirceSupport with RemoteEnvironm
         processName,
         isFragment
       )
-    invokeForSuccess(
-      HttpMethods.POST,
-      List("migrate"),
-      Query.Empty,
-      HttpEntity(migrateScenarioRequest.asJson.noSpaces),
-      List.empty
-    )
+
+    val f: EitherT[Future, NuDesignerError, Unit] = for {
+      remoteNuVersion <- EitherT(fetchRemoteNuVersion)
+
+      localNuVersion = BuildInfo.version
+
+      versionComparisonResultE = migrationApiAdapterService.compareNuVersions(localNuVersion, remoteNuVersion.value)
+      versionComparisonResult <- EitherT(Future.successful(versionComparisonResultE))
+
+      decidedMigrateScenarioRequest = migrationApiAdapterService.decideMigrationRequestDto(
+        migrateScenarioRequest,
+        versionComparisonResult
+      )
+
+      versionTag = decidedMigrateScenarioRequest match {
+        case _: MigrateScenarioRequestV1_14 => "V1_14"
+        case _: MigrateScenarioRequestV1_15 => "V1_15"
+      }
+
+      res <- EitherT(
+        invokeForSuccess(
```

Comment:
we have a Tapir endpoint, so we should use it here in the Tapir's way (https://tapir.softwaremill.com/en/latest/client/sttp.html)

Reply:
Yeah, but OTOH it is Akka-specific file. So I would rewrite it to all the way Tapir while rewriting `RemoteEnvironmentResources` into Tapir.